### PR TITLE
Issue 21 - Added configuration feature to SSG

### PR DESCRIPTION
**Pull request by irenejoeunpark**
Issue #21

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
 		"chalk": "^4.1.2",
 		"clear": "^0.1.0",
 		"figlet": "^1.5.2",
+		"g": "^2.0.1",
+		"package": "^1.0.1",
```
Comment (Kevan-Y) [inline package.json:27]: Remove those two package `npm uninstall g package`

File: src/index.js [hunk 2]
```diff
@@ -96,17 +99,50 @@ yargs
...
+			isFile(config.input),
+			config.lang,
+		);
+		console.log(config.output);
+	} catch (e) {
+		console.error(`\n\n${chalk.red.bold('Error:')} ${chalk.red(e)}`);
+		process.exit(1);
+	}
+}else {
+	yargs.argv.i.demandOption = true;
```
Comment (Kevan-Y) [inline src/index.js:132]: This will not work, please change to `yargs.demandOption(['input']);`

File: src/index.js [hunk 3]
```diff
@@ -96,17 +99,50 @@ yargs
...
+		return configurationCheck(argv.c);
+	});
+
+//if the ption is c or config, run applyConfig instead of below
+if(yargs.argv.c){
+	try {
+		const config = readConfig(yargs.argv.c);
+		convertToHtml(
+			config.input,
+			congig.stylesheet,
```
Comment (Kevan-Y) [inline src/index.js:121]: Fix typo

File: src/index.js [hunk 4]
```diff
@@ -96,17 +99,50 @@ yargs
...
+			config.output,
+			isFile(config.input),
+			config.lang,
+		);
+		console.log(config.output);
```
Comment (Kevan-Y) [inline src/index.js:126]: Remove debugging log

File: src/index.js [hunk 5]
```diff
@@ -96,17 +99,50 @@ yargs
...
+	});
+
+//if the ption is c or config, run applyConfig instead of below
+if(yargs.argv.c){
+	try {
+		const config = readConfig(yargs.argv.c);
+		convertToHtml(
+			config.input,
+			congig.stylesheet,
+			config.output,
```
Comment (Kevan-Y) [inline src/index.js:122]: If output doesn't exist set a default value to ./dist `config.output || './dist'`

File: src/index.js [hunk 6]
```diff
@@ -96,17 +99,50 @@ yargs
...
+	.check((argv) => {
+		return configurationCheck(argv.c);
+	});
+
+//if the ption is c or config, run applyConfig instead of below
+if(yargs.argv.c){
+	try {
+		const config = readConfig(yargs.argv.c);
+		convertToHtml(
+			config.input,
```
Comment (Kevan-Y) [inline src/index.js:120]: If input value is null throw an error like `Directory or file must exist.`

File: src/index.js [hunk 7]
```diff
@@ -96,17 +99,50 @@ yargs
...
+		alias: 'config',
+		describe: 'Folder/File configuration JSON file location',
+		type: 'string',
+		nargs: 1
+	})
+	.check((argv) => {
+		return configurationCheck(argv.c);
+	});
+
+//if the ption is c or config, run applyConfig instead of below
```
Comment (Kevan-Y) [inline src/index.js:115]: Fix comment typo

**Review by Kevan-Y — CHANGES_REQUESTED** [review]
Remove `examples/config.json` file and `newDist` folder
Update the README.md

File: package.json (hunk 1 above)
Comment (irenejoeunpark) [reply package.json]: resolved

**Pull request merged**